src/NServiceBus.Core/Unicast/Transport/TransportConnectionString.cs

```diff
@@ -33,21 +47,15 @@ public string GetConnectionStringOrRaiseError(TransportDefinition transportDefin
             return connectionString;
         }
 
-        static string ReadConnectionString(string connectionStringName)
-        {
-            var connectionStringSettings = ConfigurationManager.ConnectionStrings[connectionStringName];
-
-            return connectionStringSettings?.ConnectionString;
-        }
-
-        Func<string> GetValue = () => ReadConnectionString(DefaultConnectionStringName);
+        Func<string> GetValue;
 
+#if NET452
```

Comment:
by the amount of if-defs in this class I wonder whether it would be easier to understand to have to dedicated classes instead?